src/deprecated.jl

```diff
@@ -13,3 +13,6 @@ const DEProblem = AbstractDEProblem
 const DEAlgorithm = AbstractDEAlgorithm
 const DESolution = AbstractSciMLSolution
 const SciMLSolution = AbstractSciMLSolution
+
+# Deprecated `destats`
+@deprecate has_destats(x) has_stats(x)
```

Comment:
This will export `has_destats` even though currently it is not exported. This can be avoided by
```suggestion
@deprecate has_destats(x) has_stats(x) false
```